gradefetcher/gradefetcher.py

```diff
@@ -250,119 +264,102 @@ def grade_user(self, data, suffix=""):
         """
         Make a call to an external grader and retreive user's grade
         """
-        # Get EXTERNAL_GRADER from site configuration
-        grade_fetcher_settings = configuration_helpers.get_value("GRADE_FETCHER", "")
         # 1. If user in studio set authentication endpoint we call it
-        if self.authentication_endpoint:
-            try:
+        try:
+            grader_headers = {"Content-Type": "application/json"}
+            if self.authentication_endpoint:
                 # 2. Make call to auth endpoint and get the token
                 auth_response = requests.post(
```

Comment:
Please set timeout for all requests. 10 seconds should be a good maximum, otherwise servers would hang forever.

Reply:
@OmarIthawi Done, I added 10 seconds of timeout for all the requests